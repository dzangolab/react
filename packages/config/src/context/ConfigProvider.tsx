import React, { createContext } from "react";

import { AppConfig } from "..";

interface Properties {
  children: React.ReactNode;
  appConfig: AppConfig;
}

const configContext = createContext<AppConfig | undefined>(undefined);

const ConfigProvider = ({ children, appConfig }: Properties) => {
  return (
    <configContext.Provider value={appConfig}>
      {children}
    </configContext.Provider>
  );
};

export default ConfigProvider;
export { configContext };
