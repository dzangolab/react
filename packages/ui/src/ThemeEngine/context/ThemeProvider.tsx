import { createContext, useContext, useState } from "react";

import { getThemeConfig } from "../theme-config";

import type { DzTheme } from "../types";

const ThemeContext = createContext<{
  theme: DzTheme;
  setTheme: (theme: DzTheme) => void;
}>({
  theme: "auto",
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  setTheme: () => {},
});

export const DzThemeProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [theme, setTheme] = useState<DzTheme>(getThemeConfig().defaultTheme);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useDzThemeContext = () => {
  return useContext(ThemeContext);
};
