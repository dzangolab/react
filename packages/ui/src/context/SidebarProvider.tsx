import React, { createContext, useMemo, useState } from "react";

interface Properties {
  children: React.ReactNode;
}

interface SidebarContextProperties {
  open: boolean;
  toggleSidebar: () => void;
}

const sidebarContext = createContext<SidebarContextProperties | undefined>(
  undefined
);

const SidebarProvider = ({ children }: Properties) => {
  const [open, setOpen] = useState<boolean>(true);

  const toggleSidebar = () => {
    setOpen((previous) => !previous);
  };

  const providerValue = useMemo(() => ({ open, toggleSidebar }), [open]);

  return (
    <sidebarContext.Provider value={providerValue}>
      {children}
    </sidebarContext.Provider>
  );
};

export default SidebarProvider;
export { sidebarContext };
export type { SidebarContextProperties };
