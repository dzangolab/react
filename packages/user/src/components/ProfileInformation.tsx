import React from "react";

import "../assets/css/profileInformation.css";

import { useSelector } from "react-redux";

const ProfileInformation = () => {
  const { user } = useSelector((state: any) => state.auth);

  return (
    <div className="profile">
      <div className="profile-group">
        <label>Id</label>
        <p>{user?.id}</p>
      </div>
      <div className="profile-group">
        <label>Email</label>
        <p>{user?.email}</p>
      </div>
      <div className="profile-group">
        <label>Time Joined</label>
        <p>{user?.timeJoined}</p>
      </div>
    </div>
  );
};

export default ProfileInformation;
