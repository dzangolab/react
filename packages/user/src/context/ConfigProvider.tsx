import React, { createContext } from "react";

import { UserConfig } from "@/types/config";

interface Properties {
  children: React.ReactNode;
  config: UserConfig;
}

const configContext = createContext<UserConfig | undefined>(undefined);

const ConfigProvider = ({ children, config }: Properties) => {
  return (
    <configContext.Provider value={config}>{children}</configContext.Provider>
  );
};

export default ConfigProvider;

export { configContext };
