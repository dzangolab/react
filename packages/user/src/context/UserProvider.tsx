import React, { createContext, useEffect, useState } from "react";

import { getUserData, setUserData } from "../helpers";
import { useConfig } from "../hooks";
import {
  getUserRoles,
  isUserVerified,
  verifySessionRoles,
} from "../supertokens/helpers";
import { UserContextType, UserType } from "../types";
import { isEmailVerificationEnabled } from "../utils/emailVerificationconfig";

interface Properties {
  children: React.ReactNode;
}

const userContext = createContext<UserContextType | undefined>(undefined);

const UserProvider = ({ children }: Properties) => {
  const [user, setUser] = useState<UserType | undefined>(undefined);
  const [loading, setLoading] = useState<boolean>(true);
  const appConfig = useConfig();

  useEffect(() => {
    const getUser = async () => {
      try {
        if (
          appConfig &&
          (await verifySessionRoles(appConfig.user.supportedRoles))
        ) {
          const userInfo = await getUserData();

          if (userInfo) {
            setUser(userInfo);
          }
        }
      } catch (error) {
        //
      } finally {
        setLoading(false);
      }
    };

    getUser();
  }, []);

  const setVerifiedUserWithRoles = async (user: UserType | undefined) => {
    let roles, isEmailVerified;

    if (user) {
      roles = await getUserRoles();
      if (isEmailVerificationEnabled(appConfig)) {
        isEmailVerified = await isUserVerified();
        await setUserData({
          ...user,
          roles: roles,
          isEmailVerified: isEmailVerified,
        });
        setUser({ ...user, roles: roles, isEmailVerified: isEmailVerified });
      } else {
        await setUserData({
          ...user,
          roles: roles,
        });
        setUser({ ...user, roles: roles });
      }
    } else {
      setUser(undefined);
    }
  };

  return (
    <userContext.Provider
      value={{ user, loading, setUser: setVerifiedUserWithRoles }}
    >
      {loading ? null : children}
    </userContext.Provider>
  );
};

export default UserProvider;
export { userContext };
