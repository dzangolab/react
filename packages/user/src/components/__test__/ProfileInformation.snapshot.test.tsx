import React from "react";
import { create } from "react-test-renderer";
import { expect, test, vi } from "vitest";

import { userContext } from "../../context/UserProvider";
import ProfileInformation from "../ProfileInformation";

import type {
  ReactTestRenderer,
  ReactTestRendererJSON,
} from "react-test-renderer";

function toJson(component: ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const values = {
    user: { id: "5", email: "test@gmail.com", timeJoined: 320 },
    loading: false,
    setUser: vi.fn(),
  };

  const component = create(
    <userContext.Provider value={values}>
      <ProfileInformation />
    </userContext.Provider>
  );

  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
});
