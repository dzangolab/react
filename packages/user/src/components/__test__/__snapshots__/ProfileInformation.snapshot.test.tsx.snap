// Vitest Snapshot v1

exports[`Component matches snapshot 1`] = `
<div
  className="profile"
>
  <div
    className="profile-group"
  >
    <label>
      Id
    </label>
    <p>
      5
    </p>
  </div>
  <div
    className="profile-group"
  >
    <label>
      Email
    </label>
    <p>
      test@gmail.com
    </p>
  </div>
  <div
    className="profile-group"
  >
    <label>
      Time Joined
    </label>
    <p>
      320
    </p>
  </div>
</div>
`;
